hal/reset_reason_api.h

```diff
@@ -0,0 +1,67 @@
+/** \addtogroup hal */
```

Comment:
That should be guarded by device_has macro eg `DEVICE_RESET_REASON` I'd rather not make it mandatory API.